Yatima/Datatypes/Univ.lean

```diff
@@ -128,12 +128,8 @@ def instBulkReduce (substs : List Univ) : Univ → Univ
     | _ => .imax (instBulkReduce substs a) b'
   | .var nam idx => match substs.get? idx with
     | some u => u
-    -- TODO: It is still unclear, at this point, whether we should shift or
-    -- not the other variables. In fact, it is still unclear whether
-    -- this case could happen at all. It would appear that the `substs`
-    -- variable is a complete environment for the free variables
-    -- inside `univ`
-    | none => .var nam (idx - substs.length - 1) -- is this right?
+    -- This should never happen. Should we wrap this function in a monad error to avoid the exception?
+    | none => unreachable!
```

Comment:
Yeah we shouldn't have code that can panic in the typechecker

Reply:
I don't particularly mind this, because a `panic` is essentially the same as wrapping the code around a monad error -- I would even say they are isomorphic. The advantage of the monad error is that it is way easier to debug. The disadvantage is that it clutters the code and is way less efficient, by wrapping the return type in another constructor and adding checks at every function call. Exceptions are triggered by interrupts, so it doesn't require the wrapping of the return type nor these checks at function calls

Reply:
Won't this panic turn into an open reference that may compromise its compilation (when we compile the typechecker with our compiler)?

Reply:
Oh, you're worried about the compilation of the type checker itself. Well, I'm almost sure exceptions in Lean, at the AST level, are represented by constants with the value set to the default value (that's why only types that are inhabited are allowed to have exceptions). If so, then our compiler will set the unreachable exception to the default value

Reply:
Anyway, I've removed the exception in favor of a more sensible definition of `instBulkReduce`, that shifts the free variables that were left after the instantiation. This shifting case should never happen inside the typechecker because we always bulk instantiate with a complete environment though (though we could see a use of bulk instatiation with an incomplete environment)